pkgs/by-name/dy/dyalog/package.nix

```
@@ -143,8 +143,11 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   # Register some undeclared runtime dependencies to be patched in by autoPatchelfHook
+  # Note: dyalog.rt is used internally to run child APL processes in
   preFixup = ''
-    patchelf ${dyalogHome}/dyalog --add-needed libncurses.so
+    for exec in "dyalog" "dyalog.rt"; do
+        patchelf ${dyalogHome}/$exec --add-needed libncurses.so
+    done
```

```suggestion
    patchelf ${dyalogHome}/{dyalog,dyalog.rt} --add-needed libncurses.so
```
Removing the `for` loop makes the diff a bit more minimal. Up to you if you want to change it, as this somewhat decreases code legibility.

---

I'll keep it like this, because some other binaries might be added to this list in the future.